src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js

```diff
@@ -208,13 +208,16 @@ const filterCells = (cellSets, selectedCellSet) => {
 const generateData = (cellSets, selectedCellSet, embeddingData) => {
   const newCellSets = filterCells(cellSets, selectedCellSet);
 
-  const test = newCellSets.filter((d) => d.cellId < embeddingData.length).map((data) => (
-    {
-      ...data,
-      x: embeddingData[data.cellId][0],
-      y: embeddingData[data.cellId][1],
-    }
-  ));
+  const test = newCellSets
+    .filter((d) => d.cellId < embeddingData.length)
+    .filter((data) => embeddingData[data.cellId]) // filter out cells removed in data processing
+    .map((data) => (
+      {
+        ...data,
```

Comment:
since `data.cellId` is not used in the plot's spec, do you think it makes sense to remove it also from the generated data?